tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -755,6 +755,97 @@ def test_whisper_timestamp_prediction(self):
             },
         )
 
+    @slow
+    @require_torch
+    def test_whisper_large_timestamp_prediction(self):
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
+        array = np.concatenate(
+            [ds[40]["audio"]["array"], ds[41]["audio"]["array"], ds[42]["audio"]["array"], ds[43]["audio"]["array"]]
+        )
+        pipe = pipeline(
+            model="openai/whisper-large-v3",
+            return_timestamps=True,
+        )
```

Comment:
With `return_timestamps=True`, we compute the segment-level timestamps using Whisper. These timestamps were working prior to this PR. It's only the word-level timestamps that were broken. It would be good to do one check with word-level timestamps to check we get the right outputs, i.e.
```suggestion
        pipe = pipeline(
            model="openai/whisper-large-v3",
            return_timestamps="word",
        )
```

Reply:
Ah ok I see, aren't we already checking this with the second test `test_whisper_large_word_timestamps_batched' I added? 

This current test won't work if we change `return_timestamps=True` to 'word' as it won't predict full sentences anymore. I can remove it and leave only the second one.

Reply:
For the second test, you're enabling chunking `chunk_length_s=3`, which in turn sets the `stride` parameter. The case we're fixing with this PR is when we have **no chunking**, meaning `stride` is None and we have to set `num_frames` based on the `segment_size`:
https://github.com/huggingface/transformers/blob/24a340a980ec4435bc29b90552822322efa54565/src/transformers/pipelines/automatic_speech_recognition.py#L493-L494

=> hence, we should have a test where we set `chunk_length_s=None` and `return_timestamps="word"`. I don't fully understand why it doesn't work if we set `return_timestamps="word"` - could you give an example?

Reply:
> I don't fully understand why it doesn't work if we set return_timestamps="word" - could you give an example?

In the first test I added, the EXPECTED_OUTPUT contains full sentences with timestamps. If I change `return_timestamps=True`to "word", we would then predict word level timestamps and it wouldn't match with the EXPECTED_OUTPUT full sentences.  

Also, if we change `return_timestamps` to "word"`, we go back to the second test, which evaluates the behaviour of `whisper-large` for both batch_size=1 and batch_size=2 I think ;)